### PR TITLE
[ICDS Dashboard]: Hotfix in the PPD Api in case no rows are present

### PR DESCRIPTION
**Pull request by ShubhamBansal1997**

### CODE REVIEW

File: custom/icds_reports/reports/poshan_progress_dashboard_data.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def calculate_aggregated_row(data, aggregation_level, data_period, unique_id):
     aggregated_row = calculate_percentage_single_row(deepcopy(aggregated_row))
     # rounding values
     for col in ['num_launched_districts', 'num_launched_blocks', 'num_launched_states']:
-        aggregated_row[col] = round(aggregated_row[col])
+        aggregated_row[col] = round(aggregated_row.get(col, 0))
     aggregated_row = prepare_structure_aggregated_row(deepcopy(aggregated_row),
```
Comment (Rohit25negi) [inline custom/icds_reports/reports/poshan_progress_dashboard_data.py:58]: you sure, subsequent use of `aggregated_row` would not break with the same error?

**Review by Rohit25negi — APPROVED** [review]

File: custom/icds_reports/reports/poshan_progress_dashboard_data.py (hunk 1 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/reports/poshan_progress_dashboard_data.py:58]: offlined about this

File: custom/icds_reports/reports/poshan_progress_dashboard_data.py [hunk 2]
```diff
@@ -20,14 +20,15 @@
...
+        den = row.get(v[1], 1)  # to avoid 0/0 division error
         extra_number = v[2] if len(v) > 2 else None
         row[k] = calculate_percent(num, den, extra_number, truncate_out)
         # calculation is done on decimal values
         # and then round off to nearest integer
-        row[v[0]] = round(row[v[0]])
-        row[v[1]] = round(row[v[1]])
+        # and if not present defaulting them to zero
+        row[v[0]] = round(row.get(v[0], 0))
+        row[v[1]] = round(row.get(v[1], 0))
```
Comment (Rohit25negi) [inline custom/icds_reports/reports/poshan_progress_dashboard_data.py:31]: What wil happen for those values which are numbers not percentage in the respone

**Review by Rohit25negi — APPROVED** [review]

File: custom/icds_reports/reports/poshan_progress_dashboard_data.py (hunk 2 above)
Comment (ShubhamBansal1997) [reply custom/icds_reports/reports/poshan_progress_dashboard_data.py:31]: off-lined about this

**Pull request merged**